src/wamr/WAMRWasmModule.cpp

```diff
@@ -231,51 +253,18 @@ int WAMRWasmModule::executeWasmFunction(const std::string& funcName)
                      boundFunction);
         throw std::runtime_error("Did not find named wasm function");
     }
-
     // Note, for some reason WAMR sets the return value in the argv array you
     // pass it, therefore we should provide a single integer argv even though
     // it's not actually used
     std::vector<uint32_t> argv = { 0 };
-
-    // Invoke the function
-    bool success = aot_create_exec_env_and_call_function(
-      reinterpret_cast<AOTModuleInstance*>(moduleInstance),
-      reinterpret_cast<AOTFunctionInstance*>(func),
-      0,
-      argv.data());
-
+    bool success = wasm_runtime_call_wasm(execEnv, func, 0, argv.data());
     uint32_t returnValue = argv[0];
 
-    // Check function result
```

Comment:
Luckily, WAMR's error checking has improved since we first wrote this code, so we don't need such cumbersome checks.